sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -552,6 +552,34 @@ case class Substring(str: Expression, pos: Expression, len: Expression)
 }
 
 /**
+ * Returns string, with the first letter of each word in uppercase,
+ * all other letters in lowercase. Words are delimited by whitespace.
+ */
+case class InitCap(child: Expression) extends UnaryExpression with ExpectsInputTypes {
+  override def dataType: DataType = StringType
+
+  override def inputTypes: Seq[DataType] = Seq(StringType)
+
+  override def nullSafeEval(string: Any): Any = {
+    if (string.asInstanceOf[UTF8String].getBytes.length == 0) {
+      UTF8String.fromString(string.toString)
+      return string.asInstanceOf[UTF8String]
+    }
+    else {
+      val sb = new StringBuffer()
```

Comment:
The code maybe not suitable to move to UTF8String,
1. The current implementation is not based on bytes.
2. It is hard to list all the Europe special char(s) usage in bytes implementation.